wp-content/plugins/core/src/Theme/Media/Image_Wrap.php

```diff
@@ -7,19 +7,19 @@ class Image_Wrap {
 	/**
 	 * Customize WP non-captioned image output
 	 *
-	 * @param $html
-	 *
-	 * @return mixed
+	 * @param string $html
 	 *
 	 * @filter the_content
+	 *
+	 * @return string
 	 */
-	public function customize_wp_image_non_captioned_output( $html ) {
+	public function customize_wp_image_non_captioned_output( string $html ): string {
 
 		if ( ! is_singular() && ! in_the_loop() && ! is_main_query() ) {
 			return $html;
 		}
 
-		return preg_replace_callback( '/<p>((?:.(?!p>))*?)(<a[^>]*>)?\s*(<img[^>]+>)(<\/a>)?(.*?)<\/p>/i', static function ( $matches ) {
+		$result = preg_replace_callback( '/<p>((?:.(?!p>))*?)(<a[^>]*>)?\s*(<img[^>]+>)(<\/a>)?(.*?)<\/p>/i', static function ( $matches ) {
```

Comment:
This seems like a pretty big bug. Might need to look into the implications here.

Reply:
Shouldn't be, I just added some defensive coding because technically `preg_replace_callback` COULD return an array, string or null, and this way if it fails, it'll just return the original content instead.